carsus/io/nuclear/nndc.py

```diff
@@ -0,0 +1,87 @@
+import pandas as pd
+import os
+
+DECAY_DATA_SOURCE_DIR = "/home/syzygy/Desktop/TARDIS/carsus-data-nndc-main/csv"
+
+DECAY_DATA_FINAL_DIR = "/home/syzygy/Downloads/tardis-data/decay-data"
+
+
+class NNDCReader:
+
+    def __init__(self, fname=None):
+        if fname is None:
+            self.fname = DECAY_DATA_SOURCE_DIR
+        else:
+            self.fname = fname
+
+        self._decay_data = None
+
+        # TODO: create an 'nndc_columns' list present in the final dataset.
+
+    @property
+    def decay_data(self):
+        if self._decay_data is None:
+            self._decay_data = self._prepare_nuclear_dataframes()
+        return self._decay_data
+
+    def _get_nuclear_decay_dataframe(self, fpath=None):
+        all_data = []
+        for fileName in os.listdir(fpath):
+            file = os.path.join(fpath, fileName)
+
+            # convert every csv file to Dataframe and appends it to all_data
+            if os.path.splitext(file)[1] == ".csv" and os.path.getsize(file) != 0:
+                data = pd.read_csv(
+                    file,
+                )
+                all_data.append(data)
+
+        decay_data = pd.concat(all_data)
+        return decay_data
+
+    def _set_group_true(self, group):
+        # set the entire Metastable column to True if any of the values in the group is True
+        if group['Metastable'].any():
+            group['Metastable'] = True
+        return group
+
+    def _add_metastable_column(self, decay_data=None):
+        metastable_df = decay_data if decay_data is not None else self.decay_data.copy()
+
+        # Create a boolean metastable state column before the 'Decay Mode' column
+        metastable_df.insert(7, "Metastable", False)
+
+        metastable_filters = (metastable_df["Decay Mode"] == "IT") & (metastable_df["Decay Mode Value"] != 0.0) & (
+                metastable_df["Parent E(level)"] != 0.0)
+
+        metastable_df.loc[metastable_filters, 'Metastable'] = True
+
+        # avoid duplicate indices since metastable_df is a result of pd.concat operation
+        metastable_df = metastable_df.reset_index()
+
+        # Group by the combination of these columns
+        group_criteria = ['Parent E(level)', 'T1/2 (sec)', 'Isotope']
+        metastable_df = metastable_df.groupby(group_criteria).apply(self._set_group_true)
```

Comment:
I do not see where `_set_group_true` is defined.

Reply:
It is a method defined in line 42 of the same file.

https://github.com/tardis-sn/carsus/blob/424b05dc58e18c2cca5dd6e79744a6d84bb4ce45/carsus/io/nuclear/nndc.py#L42-L46

Reply:
ahah! Thank you.